fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/implementation/HashJoinImplementationRule.java

```diff
@@ -56,6 +56,10 @@ public List<OptExpression> transform(OptExpression input, OptimizerContext conte
                 joinOperator.getLimit(),
                 joinOperator.getPredicate(),
                 joinOperator.getProjection());
+        PhysicalPropertySet outputProperty = input.getOutputProperty();
+        if (outputProperty.getDistributionProperty().isAny()) {
```

Comment:
ImplementationRule is executed before extracting best plan. so may this throw NPE?  If we setting ouputProperty in extract plan phase, we may directly setting `canShuffleOutput` flag in building joinNode phase?

Reply:
thanks. @packy92 fixed, moved code to fragment builder.